indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/common/OrderedPartitionableRecord.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream.common;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a generic record with a PartitionIdType (partition id) and SequenceOffsetType (sequence number) and data
+ * from a Kafka/Kinesis stream
+ *
+ * @param <PartitionIdType> partition id
+ * @param <SequenceOffsetType>  sequence number
+ */
+public class OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType>
+{
+  public static final String END_OF_SHARD_MARKER = "EOS";
```

Comment:
added explanation. This marker is used to indicate a closed Kinesis shard (happens when the number of shards is changed). So when the indexing task sees an OrderedPartitionableRecord with sequence number "EOS" it knows to stop reading from the shard. Since this marker is used exclusively in OrderedPartitionableRecord, I feel like it should be left here.

Reply:
Hmm, I don't think this is exclusively used in `OrderedPartitionableRecord`. It's rather being used in `SeekableStreamSupervisor` and `SeekableStreamIndexTaskRunner`. Since both classes give an abstraction of common behavior of kinesis and kafka indexing services, it's not fit to use this variable in those classes which is useful only for kinesis. Also, it's confusing because its type is `String` while the offset type of kafka indexing service is `Long` but they're compared in the above classes.

What I'm suggesting is adding some more interfaces to `SeekableStreamSupervisor` and `SeekableStreamIndexTaskRunner` to leave the checking the given shard is closed or not to the implementations. What do you think?

Reply:
Yeah that makes sense. Will change that.

Reply:
moved to `KinesisSequenceNumber`